sdk/metric/sdk.go

```diff
@@ -665,15 +551,16 @@ func (m *SDK) checkpoint(ctx context.Context, descriptor *export.Descriptor, rec
 // RecordBatch enters a batch of metric events.
 func (m *SDK) RecordBatch(ctx context.Context, ls api.LabelSet, measurements ...api.Measurement) {
 	for _, meas := range measurements {
-		meas.InstrumentImpl().RecordOne(ctx, meas.Number(), ls)
+		meas.SynchronousImpl().RecordOne(ctx, meas.Number(), ls)
 	}
 }
 
-// GetDescriptor returns the descriptor of an instrument, which is not
-// part of the public metric API.
-func (m *SDK) GetDescriptor(inst metric.InstrumentImpl) *export.Descriptor {
-	if ii, ok := inst.(*instrument); ok {
-		return ii.descriptor
+// GetDescriptor returns a pointer to the descriptor of an instrument,
+// which is not part of the public metric API.  This is for testing.  Use
```

Comment:
If this is for testing and we don't want general users of the package to have access to this shouldn't we not export this?

It seems a bit like we should be dogfooding our design decision here. If we think users won't need this but we need this in other packages it seems like it might be a design problem or a testing scope problem.

Reply:
This already existed and is only used for testing.  If I moved the test into the same package, we could avoid exposing this. It's used in the stress test and at this moment I'd rather not think about how to eliminate it.

Reply:
Sounds reasonable. +1 for moving forward.

Reply:
Created https://github.com/open-telemetry/opentelemetry-go/issues/570 to track this.